#!/usr/bin/env python
PACKAGE = "bumblebee_reconfigure"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("view_o3d", bool_t, 0, "A Boolean parameter", True)
gen.add("view_publish_pointcloud", bool_t, 0, "A Boolean parameter", True)
gen.add("subscribe_topic_name", str_t, 0, "A string parameter", "/itops_f07/camera/points")
gen.add("publish_topic_name", str_t, 0, "A string parameter", "TestPC_PC2")
gen.add("distance_topic_name", str_t, 0, "A string parameter", "bumblebee/distance")
gen.add("publish_frame_id", str_t, 0, "A string parameter", "base_laser")
gen.add("marker_topic_name", str_t, 0, "A string parameter", "bounding_box_marker")
gen.add("range_max_x", double_t, 0, "A double parameter", 1, 0, 10)
gen.add("range_max_y", double_t, 0, "A double parameter", 1, 0, 10)
gen.add("range_max_z", double_t, 0, "A double parameter", 0, 0, 10)
gen.add("range_min_x", double_t, 0, "A double parameter", -1, -10, 0)
gen.add("range_min_y", double_t, 0, "A double parameter", -1, -10, 0)
gen.add("range_min_z", double_t, 0, "A double parameter", 0, -10, 0)

exit(gen.generate(PACKAGE, "bumblebee_reconfigure", "Ld06"))
